universal-application-tool-0.0.1/app/services/program/BlockDefinition.java

```diff
@@ -0,0 +1,48 @@
+package services.program;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+
+/** Defines a single program block, which contains a list of questions and data about the block. */
+@AutoValue
+public abstract class BlockDefinition {
+
+  public abstract String name();
+
+  public abstract String description();
+
+  public abstract Optional<Predicate> hidePredicate();
+
+  public abstract Optional<Predicate> optionalPredicate();
+
+  public abstract ImmutableList<String> questionDefinitions();
```

Comment:
Suggestion (blocking): [QuestionDefinition] is available now so this can use the [real thing](https://github.com/seattle-uat/universal-application-tool/blob/main/universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java).

Reply:
done!